lib/services/oidc.go

```diff
@@ -299,6 +299,16 @@ func (o *OIDCConnectorV2) SetResourceID(id int64) {
 	o.Metadata.ID = id
 }
 
+// WithoutSecrets returns an instance of resource without secrets.
+func (o *OIDCConnectorV2) WithoutSecrets() Resource {
+	if o.GetClientSecret() == "" {
+		return o
+	}
+	o2 := *o
```

Comment:
same as above, `OIDCConnectorSpecV2` has some slice fields that won't be deep-copied